wcfsetup/install/files/lib/system/flood/FloodControl.class.php

```diff
@@ -0,0 +1,241 @@
+<?php
+namespace wcf\system\flood;
+use wcf\data\object\type\ObjectTypeCache;
+use wcf\system\SingletonFactory;
+use wcf\system\WCF;
+use wcf\util\DateUtil;
+use wcf\util\UserUtil;
+
+/**
+ * Flood control tracks the times contents are created by users.
+ * 
+ * @author      Matthias Schmidt
+ * @copyright   2001-2020 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package     WoltLabSuite\Core\System\Flood
+ * @since       5.4
+ */
+final class FloodControl extends SingletonFactory {
+	private const PRUNE_TIME = TIME_NOW - 31 * 86400;
+	private const IDENTIFIER_PREFIX = WCF_UUID . ':' . self::class . ':';
+	
+	/**
+	 * Returns the number of contents by a certain identifier type within a certain period of
+	 * time `($time-$interval, $time]` and the earliest time within the period at which content
+	 * was created.
+	 */
+	private function countContentByIdentifier(string $objectType, string $identifier, \DateInterval $interval, int $time): array {
+		$sql = "SELECT  COUNT(*) AS count, MIN(time) AS earliestTime
+			FROM    wcf" . WCF_N . "_flood_control
+			WHERE   objectTypeID = ?
+			        AND identifier = ?
+			        AND (time > ? AND time <= ?)";
+		$statement = WCF::getDB()->prepareStatement($sql);
+		$statement->execute([
+			$this->getObjectTypeID($objectType),
+			$identifier,
+			DateUtil::getDateTimeByTimestamp($time)->sub($interval)->getTimestamp(),
+			$time,
+		]);
+		
+		return $statement->fetchSingleRow();
+	}
+	
+	/**
+	 * Returns the number of contents the active user created of the given type within a certain
+	 * period of time `($time-$interval, $time]` and the earliest time within the period content
+	 * was created.
```

Comment:
Adjust the phrasing as well. Or maybe simply refer back to `countContentByIdentifier`? Not sure which method should be documented, either `countContentByIdentifier` or the public one (this).

Same applies to all the other methods.